fix + ♻️ #4443 navbar hover + refactor

petersopko opened the pull request:
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).

👇 \_ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring

## Context

- [x] closes #4443, closes #4451, closes #4475
- [ ] Requires deployment <>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR
![image](https://user-images.githubusercontent.com/73316633/206045021-53b6b33c-b076-481a-80e1-16dfc62f7079.png)


#### Optional

- [ ] I've tested it at </bsx/collection>
- [x] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=<My_Kusama_Address_check_https://github.com/kodadot/nft-gallery/blob/main/CONTRIBUTING.md#creating-your-ksm-address>)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

https://user-images.githubusercontent.com/73316633/206044795-8b5e7ca9-c68d-4ab1-a8d7-6cd70b95c81c.mp4

petersopko commented:
had to move away from native buefy navbar dropdown to custom dropdown due to known bug in: 
- https://github.com/buefy/buefy/issues/3086

petersopko commented:
I'm not sure if this is tracked elsewhere but there's a lot of funky stuff going on with mobile menu, such as:

1. you have to double tap to cancel selected option
2. now the UI is a bit off and needs rethinking, it would be nice to have unified look for create/stats and chain
3. connect button basically hidden for me on iPhone, possible to scroll down to it in dev view on chrome

I'd like to start working on these tomorrow.

yangwao commented:
> if I click the menu, the dropdown still stays open

yup, same 

<img width="343" alt="image" src="https://user-images.githubusercontent.com/5887929/206203855-48c5d307-5934-4465-93b9-734fb41e8f8c.png">

petersopko commented:
@preschian @yangwao fixed the click/hover issue, but I just found out that on mobile, navbar is with this PR almost unusable, let me fix that here as well. I'll try to finish it today.

yangwao commented:
what's happening here

<img src="https://media2.giphy.com/media/11dmEI4SVjTKPm/giphy.gif"/>

petersopko commented:
> what's happening here
> 
> ![](https://camo.githubusercontent.com/60f4a2550527723315b0e3fe4bb7bef299b419472d9e93b3356a9a2871884594/68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f3131646d45493453566a544b506d2f67697068792e676966) [ ![68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f3131646d45493453566a544b506d2f67697068792e676966](https://camo.githubusercontent.com/60f4a2550527723315b0e3fe4bb7bef299b419472d9e93b3356a9a2871884594/68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f3131646d45493453566a544b506d2f67697068792e676966) ](https://camo.githubusercontent.com/60f4a2550527723315b0e3fe4bb7bef299b419472d9e93b3356a9a2871884594/68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f3131646d45493453566a544b506d2f67697068792e676966) [ ](https://camo.githubusercontent.com/60f4a2550527723315b0e3fe4bb7bef299b419472d9e93b3356a9a2871884594/68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f3131646d45493453566a544b506d2f67697068792e676966)

I have just noticed that @prachi00 has taken this
- https://github.com/kodadot/nft-gallery/issues/4125

which seems to be solving mobile navbar altogether. 

I think if this should be only desktop, it's good to go.

petersopko commented:
one more small thing

exezbcz commented:
I guess we misunderstood each other. I believe the hover dropdown should only be shown on the explore button, as it was intended. The problem is that you've already finished it.

Solution is to remove the hover dropdown on network - including the arrow - and leave it only on the collection, because the button should also be clickable - redirect to kodadot/explore and the dropdown is only for extensive list, so you don't have to switch later on kodadot.xyz/explore/something.

Then there's the not-so-ux-friendly solution of putting everything on hover dropdown and adding a down pointing chevron.

Sorry for not making it clearer earlier.
thoughts? @yangwao @petersopko

petersopko commented:
@exezbcz 

- removed chevron from chain select
- only explore is triggered on hover, rest is triggered with click
- clicking on explore is redirecting to `/${urlPrefix}/explore`

lmk if it's ok

exezbcz commented:
> @exezbcz
> 
> * removed chevron from chain select
> * only explore is triggered on hover, rest is triggered with click
> * clicking on explore is redirecting to `/${urlPrefix}/explore`
> 
> lmk if it's ok

Niceee, quick rework, i think now its working nicely. I just realised that we have also dropdown on hover on profile and wallet, i am dumb but i think since we are going to update the wallet anyway i think its fine. 

Soooo, i believe it will change soon, but for now, this works for me.

yangwao commented on components/Navbar.vue:
```diff
@@ -48,120 +48,21 @@
+      <NavbarStats
+        class="navbar-stats custom-navbar-item"
+        data-cy="stats"
+        :is-bsx="isBsx"
```
if another chain is coming, will we be adding line by line? 
I guess we should look for more functional and future proof way

yangwao commented:
>   ![image](https://user-images.githubusercontent.com/73316633/206045021-53b6b33c-b076-481a-80e1-16dfc62f7079.png)

This is a bit antipattern, but probably a good recommendation would be to reuse what's @roiLeo is doing in #4430

petersopko commented:
> > ![image](https://user-images.githubusercontent.com/73316633/206045021-53b6b33c-b076-481a-80e1-16dfc62f7079.png)
> 
> This is a bit antipattern. Probably a good recommendation/suggestion would be to reuse pattern what's @roiLeo is doing in #4430 Also prefix Navbar* is not needed if it's in folder navbar

I've fixed the naming and the `isChain` was replaced by `chain` which is resolved inside the dropdowns themselves (no need to add new props for new chains)

As for the mentioned antipattern, I've checked out @roiLeo's WIP on NeoDropdowns and I'm not sure what do you mean exactly. 

If it's regarding some of the code repetition, I've split the navbar into multiple components for readability and replaced the native navbar dropdowns due to https://github.com/buefy/buefy/issues/3086.

Maybe we want to refactor this to final version once #4430 is done? My understanding is that `NeoDropdown` and `NeoDropdownItem.vue` will replace the Buefy dropdown components. I can imagine this being simplified then.

petersopko commented:
also, https://github.com/kodadot/nft-gallery/issues/4475 is fixed here

yangwao commented:
Yes, let's look at when #4430 will be ready 

Looking on #4125 will fix some residues 

pay 70 usd

yangwao commented:
pay 70 usd

yangwao commented:
😍 Perfect, I’ve sent the payout 
💵 $70 @ 30.32 USD/KSM ~ 2.309 $KSM
🧗 [FNyt7T1xdbhN7dagf7yGYCRJuE3R45VmTCE3tjzy1rKxa7y](https://kusama.subscan.io/account/FNyt7T1xdbhN7dagf7yGYCRJuE3R45VmTCE3tjzy1rKxa7y)
🔗 [0x5beaf50340e8543cff44079b11227c078526f2a894ad6aca10ae57fa2bb01284](https://kusama.subscan.io/extrinsic/0x5beaf50340e8543cff44079b11227c078526f2a894ad6aca10ae57fa2bb01284)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)
